scripts/west_commands/runners/jlink.py

```diff
@@ -25,7 +26,33 @@
 except ImportError:
     MISSING_REQUIREMENTS = True
 
-DEFAULT_JLINK_EXE = 'JLink.exe' if sys.platform == 'win32' else 'JLinkExe'
+if sys.platform == 'win32':
+    # JLink.exe can collide with the JDK executable of the same name
+    # Look in the usual locations before falling back to $PATH
+    for path in [Path(r"C:\Program Files\SEGGER"), Path(r"C:\Program Files (x86)\SEGGER")]:
```

Comment:
There are [environment variables](https://en.wikipedia.org/wiki/Environment_variable#Windows:~:text=%25ProgramFiles%25%2C%20%25ProgramFiles(x86)%25%2C%20%25ProgramW6432%25) for the Program Files.